servo/repeating.py

```diff
@@ -62,13 +71,28 @@ def start_repeating_task(
 
         async def repeating_async_fn() -> None:
             while True:
-                callable()
-                await asyncio.sleep(every.total_seconds())
+                t0 = time.time()
+                if asyncio.iscoroutinefunction(function):
+                    await function()
+                elif callable(function):
+                    function()
+                else:
+                    raise TypeError(f"function={function} must be Awaitable or Callable, but has "
+                        f"type(function)={type(function)}.")
+                t1 = time.time()
+                sleep_time = every.total_seconds()
+                if time_correction:
+                    sleep_time -= max(t1 - t0, 0)
+                if sleep_time > 0:
+                    await asyncio.sleep(sleep_time)
+                else:
+                    await asyncio.sleep(float_info.epsilon)
```

Comment:
Please add a comment indicating this is a concurrency failsafe (assuming that was the reason it was added)